### PR TITLE
Add integration test cases

### PR DESCRIPTION
**Pull request by jhs7jhs**
Adding test cases according to #761.
This should be merged after #912.

### CODE REVIEW

**Comment by sgkim126** [comment]
I moved chain, discovery2, discovery5 and transactions to integration.long directory.
And I split some tests. https://github.com/CodeChain-io/codechain/commit/d37ad2cbbc4188347dfbca07ed6ca1f40a136bb5

File: test/src/integration/chain.test.ts [hunk 1]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
             });
     });
 
-    // Not implemented
-    it("getNumberOfShards");
-    it("getShardRoot");
+    it("getNumberOfShards", async function() {
+        expect(
+            await node.sdk.rpc.sendRpcRequest("chain_getNumberOfShards", [null])
+        ).not.to.be.null;
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]:
```ts
).to.equal(1);
```

File: test/src/integration/chain.test.ts [hunk 2]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
...
+
+        expect(
+            await node.sdk.rpc.sendRpcRequest("chain_getNumberOfShards", [0])
+        ).not.to.be.null;
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]:
```ts
).to.equal(1);
```

File: test/src/integration/chain.test.ts [hunk 3]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
...
+    });
+
+    it("getShardRoot", async function() {
+        await node.sdk.rpc
+            .sendRpcRequest("chain_getShardRoot", [0, null])
+            .then(result => {
+                try {
+                    if (result !== null) {
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]: Is it okay when `chain_getShardRoot` returns `null`?

File: test/src/integration/chain.test.ts [hunk 4]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
...
+                } catch (e) {
+                    expect.fail();
+                }
+            });
+
+        await node.sdk.rpc
+            .sendRpcRequest("chain_getShardRoot", [0, 0])
+            .then(result => {
+                try {
+                    if (result !== null) {
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]: Is it okay when `chain_getShardRoot` returns `null`?

File: test/src/integration/chain.test.ts [hunk 5]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
...
+                } catch (e) {
+                    expect.fail();
+                }
+            });
+
+        await node.sdk.rpc
+            .sendRpcRequest("chain_getShardRoot", [10000, null])
+            .then(result => {
+                try {
+                    if (result !== null) {
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]: Is it okay when `chain_getShardRoot` doesn't return `null`?

File: test/src/integration/chain.test.ts [hunk 6]
```diff
@@ -413,9 +413,67 @@ describe("chain", function() {
...
+                }
+            });
+    });
+
+    it("getMiningReward", async function() {
+        await node.sdk.rpc
+            .sendRpcRequest("chain_getMiningReward", [0])
+            .then(result => {
+                try {
+                    if (result !== null) {
```
Comment (sgkim126) [inline test/src/integration/chain.test.ts]: How about specifying the mining reward?
We know the scheme file.

File: test/src/integration.long/transactions.test.ts [hunk 7]
```diff
@@ -83,9 +86,35 @@ describe("transactions", function() {
...
-        // Not implemented
-        it("mint amount U64 max");
-        it("mint amount exceeds U64");
+        it("Mint unsuccessful - mint amount U64.MAX_VALUE + 1", async function() {
+            const scheme = node.sdk.core.createAssetScheme({
+                shardId: 0,
+                metadata: "",
+                amount: 0
+            });
+            (scheme.amount.value as any) = U64.MAX_VALUE.value.plus(1);
```
Comment (sgkim126) [inline test/src/integration.long/transactions.test.ts]: What's the value of `U64.MAX_VALUE.value.plus(1)`?

Comment (jhs7jhs) [reply test/src/integration.long/transactions.test.ts]: `U64` internally uses `BigNumber`, which does not have any limitation. This is the code to send `U64.MAX_VALUE + 1` by a trick.

**Pull request merged**